crypto/multikey/kind_test.go

```diff
@@ -156,15 +158,108 @@ func TestKeyKindFromSignature(t *testing.T) {
 				t.Errorf("KeyKindFromSignature() err = %v, want %v", err, tt.wantErr)
 			}
 
-			if !assert.Equal(key, tt.wantKey) {
+			if !assertion.Equal(key, tt.wantKey) {
 				t.Errorf("KeyKindFromSignature() key = %v, want %v", key, tt.wantKey)
 			}
 		})
 	}
 }
 
+func TestGetKeyKindFromBytes(t *testing.T) {
+	assertion := assert.New(t)
+	type args struct {
+		publicKey  []byte
+		privateKey []byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantKey crypto.PrivateKey
+		wantErr error
+	}{
+		{
+			name: "success-ed25519-sofia-key",
+			args: args{
+				publicKey:  ed25519test.SofiaPublicKey.Bytes(),
+				privateKey: ed25519test.SofiaPrivateKey.Bytes(),
+			},
+			wantKey: ed25519test.SofiaPrivateKey,
+		},
+		{
+			name: "success-ed25519-charlotte-key",
+			args: args{
+				publicKey:  ed25519test.CharlottePublicKey.Bytes(),
+				privateKey: ed25519test.CharlottePrivateKey.Bytes(),
+			},
+			wantKey: ed25519test.CharlottePrivateKey,
+		},
+		{
+			name: "success-secp256k1-sofia-key",
+			args: args{
+				publicKey:  secp256k1test.SofiaPublicKey.Bytes(),
+				privateKey: secp256k1test.SofiaPrivateKey.Bytes(),
+			},
+			wantKey: secp256k1test.SofiaPrivateKey,
+		},
+		{
+			name: "success-secp256k1-charlotte-key",
+			args: args{
+				publicKey:  secp256k1test.CharlottePublicKey.Bytes(),
+				privateKey: secp256k1test.CharlottePrivateKey.Bytes(),
+			},
+			wantKey: secp256k1test.CharlottePrivateKey,
+		},
+		{
+			name: "success-sr25519-sofia-key",
+			args: args{
+				publicKey:  sr25519test.SofiaPublicKey.Bytes(),
+				privateKey: sr25519test.SofiaPrivateKey.Bytes(),
+			},
+			wantKey: sr25519test.SofiaPrivateKey,
+		},
+		{
+			name: "success-sr25519-charlotte-key",
+			args: args{
+				publicKey:  sr25519test.CharlottePublicKey.Bytes(),
+				privateKey: sr25519test.CharlottePrivateKey.Bytes(),
+			},
+			wantKey: sr25519test.CharlottePrivateKey,
+		},
+		{
+			name: "err-invalid-public-key-bytes",
+			args: args{
+				publicKey:  []byte{0x1},
+				privateKey: sr25519test.CharlottePrivateKey.Bytes(),
+			},
+			wantErr: ErrNoMatch,
+		},
+		{
+			name: "err-no-key-kinds",
+			args: args{
+				publicKey:  []byte{0x1},
+				privateKey: []byte{0x2},
+			},
+			wantErr: ErrNoMatch,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			privateKey, err := GetKeyKindFromBytes(tt.args.publicKey, tt.args.privateKey)
+
+			if !(err == tt.wantErr) {
+				t.Errorf("GetKeyKindFromBytes() err = %v, want %v", err, tt.wantErr)
+			}
+
+			if !assertion.Equal(privateKey, tt.wantKey) {
+				t.Errorf("GetKeyKindFromBytes() key = %v, want %v", privateKey, tt.wantKey)
+			}
+		})
+	}
+}
+
 func TestRemoveDuplicates(t *testing.T) {
-	assert := assert.New(t)
+	assertion := assert.New(t)
```

Comment:
I have come to realise that `assert := assert.New(t)` is not a good approach. For two reasons I can see now, the first is what you have addressed; changing what assert does part the want through a function. Second is that it gives less information about the test case that failed. Can you remove this line and instead add t to the assert calls.